core/invoker/src/main/scala/whisk/core/containerpool/docker/DockerContainer.scala

```diff
@@ -250,8 +255,7 @@ class DockerContainer(id: ContainerId, ip: ContainerIp)(implicit docker: DockerA
 
           if (retries > 0 && !isComplete && !isTruncated) {
             logger.info(this, s"log cursor advanced but missing sentinel, trying $retries more times")
-            Thread.sleep(logsRetryWait.toMillis)
-            readLogs(retries - 1)
+            akka.pattern.after(logsRetryWait, as.scheduler)(readLogs(retries - 1))
```

Comment:
should we make read logs tail recursive?

Reply:
I think it now effectively is trampolined since it's decoupled via the akka scheduler. It's not really tail-recursive though.